src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const arrayStates = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData)
+    } else if (action.type === 'keysToRemove') {
```

Comment:
The condition should check for 'action.type === 'removeProperties'' instead of 'action.type === 'keysToRemove''. The latter is not a valid action type according to the task description.